internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,628 @@
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			applicationType := volumegroups.ApplicationTypeSAPNegativeHANA
+
+			volumeList, err := expandNetAppVolumeGroupVolumes(model.Volumes)
+			if err != nil {
+				return err
+			}
+
+			// Performing some basic validations that are not possible in the schema
+			if errorList := validateNetAppVolumeGroupVolumes(volumeList, applicationType); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			// Parse volume list to set secondary volumes for CRR
+			for i, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Modify volumeType as data protection type on main volumeList
+					// so it gets created correctly as data protection volume
+					(*volumeList)[i].Properties.VolumeType = utils.String("DataProtection")
+				}
+			}
+
+			// TODO: deploymentSpecId is temporary until the backend is updated and deploymentSpecId is not required anymore,
+			//       it will be handled internally by the RP
+			deploymentSpecId := "20542149-bfca-5618-1879-9863dc6767f1"
+
+			parameters := volumegroups.VolumeGroupDetails{
+				Location: utils.String(location.Normalize(model.Location)),
+				Properties: &volumegroups.VolumeGroupProperties{
+					GroupMetaData: &volumegroups.VolumeGroupMetaData{
+						GroupDescription:      utils.String(model.GroupDescription),
+						ApplicationType:       &applicationType,
+						ApplicationIdentifier: utils.String(model.ApplicationIdentifier),
+						DeploymentSpecId:      utils.String(deploymentSpecId),
+					},
+					Volumes: volumeList,
+				},
+			}
+
+			err = client.VolumeGroupsCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume group be completely provisioned
+			if err := waitForVolumeGroupCreateOrUpdate(ctx, client, id); err != nil {
+				return err
+			}
+
+			// CRR - Authorizing secondaries from primary volumes
+			for _, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Getting secondary volume resource id
+					secondaryId := volumes.NewVolumeID(subscriptionId,
+						model.ResourceGroupName,
+						model.AccountName,
+						getResourceNameString(volumeCrr.Properties.CapacityPoolResourceId),
```

Comment:
`getResourceNameString(` has the potential to crash if `volumeCrr.Properties.CapacityPoolResourceId` or `volumeCrr.Name` is nil. We should check that it isn't nil and also read that into `capacitypools.ParseCapacityPoolID(volumeCrr.Properties.CapacityPoolResourceId)` so we know for sure it's what we want.

We should remove `getResourceNameString` wherever it is used in favor of ids.

Reply:
The problem here is I need to obtain the capacity pool name from the id, if there is something on TF SDK that gives me that I would be fine, I can do the extra checks you're mentioning but I need a good way to extract the name from a resource Id before retiring getResourceNameString .

Reply:
I changed the getResourceNameString function to use pointer.From and test for empty result and return the name or empty string if the capacityPoolId is null, which this will never happen BTW since it is a required argument in the PUT request but I agree that I had to add better handling.

Reply:
I still want to get away from `getResourceNameString ` because we don't know for sure it's what we want. The following should get your capacity pool name and we will know that `volumeCrr.Properties.CapacityPoolResourceId` is a capacity pool id and if it isn't, we can error.

```
capacityId, err := capacitypools.ParseCapacityPoolID(volumeCrr.Properties.CapacityPoolResourceId)
if err != nil {
  return err
}
secondaryId := volumes.NewVolumeID(subscriptionId,
						model.ResourceGroupName,
						model.AccountName,
						capacityId.CapacityPoolName
```

Reply:
Got it, that's a good one.

Reply:
I'll be able to replace only when I have a proper Id, like the capacity pool id listed above, but I have other use-case where (unfortunately) the resource name for volume in Azure gets like this account_name/pool_name/volume_name so in this case I will still use it since there is no way to use the ParseCapacityPoolID.

Reply:
What I did for volume.Name and volumeCrr.Name cases, I renamed that function to something more specific, `getUserDefinedVolumeName` so folks don't use for the wrong reasons.